File: JS/problem010.js

// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//
// Find the sum of all the primes below two million.

((l)=>{for(s=[false,false].concat([...Array(l-2)].map(_=>true)),a=2,b=a*a;a*a<l;(s[a]&&b<l)?b+=a:a++&&(b=a*a))s[b]=false;return s;})(2e6).reduce((s,v,i)=>v?s+i:s);

File: c++/problem010.cpp
#include <iostream>
#include <vector>
#include "util.h"

// problem010.cpp
// Find the sum of all the primes below two million.

template <class T>
T sumVector(std::vector<T> &v) {
	T sum = 0;
	for (auto& i : v) { sum += i; }	
	return sum;
}

int main() {

	long limit = 2e6;
	std::vector<long> primes;
	incrementalSieve(limit, primes);
	
	std::cout << sumVector(primes) << std::endl;
	return 0;
}
